maestro-ai/src/main/java/maestro/ai/openai/Client.kt

```diff
@@ -0,0 +1,128 @@
+package maestro.ai.openai
+
+import io.ktor.client.HttpClient
+import io.ktor.client.plugins.HttpTimeout
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.client.request.post
+import io.ktor.client.request.setBody
+import io.ktor.client.statement.bodyAsText
+import io.ktor.http.ContentType
+import io.ktor.http.contentType
+import io.ktor.http.isSuccess
+import io.ktor.util.encodeBase64
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonObject
+import maestro.ai.AI
+import maestro.ai.CompletionData
+import maestro.ai.common.Base64Image
+import org.slf4j.LoggerFactory
+
+private const val API_URL = "https://api.openai.com/v1/chat/completions"
+
+private val logger = LoggerFactory.getLogger(OpenAI::class.java)
+
+class OpenAI(
+    defaultModel: String = "gpt-4o-2024-08-06",
+    private val apiKey: String,
+    private val defaultTemperature: Float = 0.2f,
+    private val defaultMaxTokens: Int = 1024,
+    private val defaultImageDetail: String = "low",
+) : AI(defaultModel = defaultModel) {
+    private val client = HttpClient {
```

Comment:
Why we use 2 different HTTP clients here for 2 models?

Reply:
thank you, fixed.

Now the `AI` superclass has a `protected httpClient` property.